wallet/core/src/account/descriptor.rs

```diff
@@ -0,0 +1,181 @@
+//!
+//! Account descriptors (client-side account information representation).
+//!
+
+use crate::derivation::AddressDerivationMeta;
+use crate::imports::*;
+use borsh::{BorshDeserialize, BorshSerialize};
+use kaspa_addresses::Address;
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+
+#[derive(Clone, Debug, Serialize, Deserialize, BorshSerialize, BorshDeserialize)]
+pub struct AccountDescriptor {
+    pub kind: AccountKind,
+    pub account_id: AccountId,
+    pub account_name: Option<String>,
+    pub prv_key_data_ids: AssocPrvKeyDataIds,
+    pub receive_address: Option<Address>,
+    pub change_address: Option<Address>,
+
+    pub properties: BTreeMap<AccountDescriptorProperty, AccountDescriptorValue>,
+}
+
+impl AccountDescriptor {
+    pub fn new(
+        kind: AccountKind,
+        account_id: AccountId,
+        account_name: Option<String>,
+        prv_key_data_ids: AssocPrvKeyDataIds,
+        receive_address: Option<Address>,
+        change_address: Option<Address>,
+    ) -> Self {
+        Self { kind, account_id, account_name, prv_key_data_ids, receive_address, change_address, properties: BTreeMap::default() }
+    }
+
+    pub fn with_property(mut self, property: AccountDescriptorProperty, value: AccountDescriptorValue) -> Self {
+        self.properties.insert(property, value);
+        self
+    }
+}
+
+#[derive(Clone, Debug, Ord, PartialOrd, Eq, PartialEq, Hash, Serialize, Deserialize, BorshSerialize, BorshDeserialize)]
+
+pub enum AccountDescriptorProperty {
+    AccountIndex,
+    XpubKeys,
+    Ecdsa,
+    DerivationMeta,
+    Other(String),
```

Comment:
This needs to be serializable and the string is meant to be for user consumption (a caption), so I think this is fine.  I think `Display` or `Cow` would overcomplicate this here.  Maybe `Cow` would make sense, need to see how this handles serializations.

Reply:
I didn't find read operation using it, yet.
But I think it will force caller to allocate string to get value, that's why I think about Cow

Reply:
Descriptors (Wallet and Account) are API representations of available instances. They are enumerated and shipped as responses or as notifications via the WalletApi. This frequently happens across async execution chains.

Then client-side they are deserialized and carry their data (as in lists of descriptors are retained).   Also not yet sure how this will be handled for WASM bindings but if such structs are retained they will need to own strings.

So I think this is not feasible, at least not right now. I think we should postpone this till Jan once the client side utilization becomes more apparent.

Reply:
Yes the structure should own string, that's clear.

That's about api.
For example if you have Hashmap<String,Whatever> and you want to get value. Here you are not able to get value by str and have to allocate string. However if it was Cow, you can get value passing the ref

Reply:
No, I am aware of how Cow functions.  Most data structures here have multi-purpose behavior tho, a lot requiring serialization and retention, which makes these typical optimizations not quite feasible.